lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+                    provided. These are equal to the weight associated with the
+                    first and last values along the weighting coord e.g.
+                    forecast_period.
+
+        """
+        source_points = weights_cube.coord(self.weighting_coord_name).points
+        target_points = cube.coord(self.weighting_coord_name).points
+        source_weights = weights_cube.core_data()
+        axis, = weights_cube.coord_dims(self.weighting_coord_name)
+
+        coord_values = (
+            {self.weighting_coord_name: lambda cell: cell == source_points[0]})
+        constr = iris.Constraint(coord_values=coord_values)
+        lower_fill_value = weights_cube.extract(constr).core_data()
+
+        coord_values = (
+            {self.weighting_coord_name:
+             lambda cell: cell == source_points[-1]})
+        constr = iris.Constraint(coord_values=coord_values)
+        upper_fill_value = weights_cube.extract(constr).core_data()
+
+        fill_value = (lower_fill_value, upper_fill_value)
+        return source_points, target_points, source_weights, axis, fill_value
+
+    @staticmethod
+    def _create_coord_and_dims_list(
+            base_cube, cube_with_exception_coord, coord_list,
+            exception_coord_name):
+        """Create a list of coordinates and their dimensions, primarily
+        for usage when constructing an iris.cube.Cube.
+
+        Args:
+            base_cube (iris.cube.Cube):
+                Cube from which all the coordinates within the coord_list
+                will be copied, apart from the coordinate defined by the
+                exception_coord_name, which will be taken from the
+                cube_with_exception_coord.
+            cube_with_exception_coord (iris.cube.Cube):
+                Cube from which the exception coordinate will be taken.
+            coord_list (iterable):
+                List/tuple of coordinates.
+            exception_coord_name (str):
+                Name of the exception coordinate that will be taken from the
+                cube_with_exception_coord.
+
+        Returns:
+            new_coord_list (list):
+                List of tuples where each tuple is of the form
+                (coordinate, index_of_coord). This list, may include the
+                exception_coord_name taken from the cube_with_exception_coord.
+                If the coordinate does not have an associated dimension e.g.
+                the coordinate is a scalar coordinate, then the index of the
+                coordinate is set equal to None.
+        """
+        new_coord_list = []
+        for coord in coord_list:
+            if (coord.name() == exception_coord_name or
+                    base_cube.coord_dims(coord) ==
+                    base_cube.coord_dims(exception_coord_name)):
+                exception_coord = cube_with_exception_coord.coord(coord.name())
+                if len(cube_with_exception_coord.coord_dims(coord)) > 0:
```

Comment:
PYLINT wants you to change this to 
if cube_with_exception_coord.coord_dims(coord):
as it thinks there is no need to actually count the list, you only care that it is not empty.
[Same again 6 lines later]

Reply:
Done.